contracts/VeYfiRewards.vy

```diff
@@ -380,18 +380,20 @@ def claim_many(_receivers: address[20]) -> bool:
 
 
 @external
-def queueNewRewards(_amount: uint256) -> bool:
+def burn(_coin: address) -> bool:
     """
-    @notice Receive 3CRV into the contract and trigger a token checkpoint
-    @param _amount Amount to transfer
+    @notice Receive YFI into the contract and trigger a token checkpoint
+    @param _coin Address of the coin being received (must be YFI)
     @return bool success
     """
-    assert _amount > 0
+    assert _coin == self.token
     assert not self.is_killed
 
-    ERC20(self.token).transferFrom(msg.sender, self, _amount)
-    if (block.timestamp > self.last_token_time + TOKEN_CHECKPOINT_DEADLINE):
-        self._checkpoint_token()
+    amount: uint256 = ERC20(_coin).balanceOf(msg.sender)
```

Comment:
should this be limited to allowance instead? or min between the two?

Not sure if amount may benefit from optional value in vyper, since you may want to burn an amount where
`amount < balanceOf()` for some reason and this method makes it so you need to have a burner address with full amount approved to this contract.